### DISCUSSION TITLE
Issues with availability after power outage

### DISCUSSION THREAD

**Discussion by dortamur** [Koenkk/zigbee2mqtt#11580] [General]
I have about 75 devices on a ZZH CC2652R controller, with mostly Xiaomi battery devices and Nue/3A powered devices (which act as routers), and a smattering of other brands (Ikea, Philips, Sercomm).

I'm having issues with device state after a power outage. That is, power is restored, but the device state remains offline/unavailable. This leads to lights actually on when they shouldn't be, and power plugs off when they shouldn't be.

In my config I have:
```yaml
advanced:
  availability:
    active:
      timeout: 15
    passive:
      timeout: 1500
```

For a few high-priority devices I have per-device:
```yaml
  availability:
    timeout: 1
```

In the Zigbee2MQTT dash, after selecting a device then the "Exposes" tab, there's a sync icon for the State property for things that are switchable. For some devices if I hit that sync icon, the device quickly updates the state, even if it's been "offline" for a day or more.

What is the equivalent MQTT command to what this sync button does?

Also with these default availability settings, even after a device is responsive, many still have `availability=offline` in MQTT.

A lot of the time, an unavailable device is just an inconvenience (presence detector not working, light being on when it shouldn't), but I put a smart plug on my refrigerator for power monitoring, set up an automation such that if it was ever not "on" to send an Alert and turn it back on - but in its "offline" state, the automation can't turn the plug back on, even though it is technically powered back on. 

Any help or insight would be appreciated. :-)

**Comment by dortamur** [comment]
Ok, I've figured out how to emulate the Z2M state ping (turned on debug logging, pressed it, and watched the logs - obvious in hindsight).
If a device like a smart plug is offline/unavailable, I can send `{"state":""}` to `zigbee2mqtt/deviceid/get`, and it wakes up the device and updates the status in Z2M.

So I can at least set up an automation to poke important plugs if they're not responding - but this still doesn't explain why the Z2M availability isn't reporting that the device is online...

**Comment by tijmenvdk** [comment]
Have you ever managed to find a cause? I had a power outage 2 days ago and am seeing this exact behavior: devices going or remaining offline for no apparent reason. I have deleted a couple of them from Zigbee2MQTT (both with and without "force remove"), but if I do that I am no longer able to re-add the devices at all. Could it be that the Conbee II stick is malfunctioning somehow as a result of the power outtage?

**Comment by janhark** [comment]
I have come to the conclusion that after a while (at least on my system running on a Slaesh stick) Zigbee2MQTT simply just stops updating the Availability information for all the devices. Restarting Zigbee2MQTT helps with this but it is certainly annoying.

Wondering if I should perhaps just schedule a nightly Z2M restart - really fixing the symptom and not the cause, but I do not have an idea about the actual cause for this.
